src/index.ts

```diff
@@ -1,128 +1,159 @@
-        return false;
-    }
-
-    async save(image: StorageBase.Image, targetDir?: string): Promise<string> {
-        console.log("save", image.name)
-        const folder = targetDir || this.getTargetDir(this.uploadOptions.folder)
-        const fileName = this.getSanitizedFileName(image.name);
-
-        const file = await readFileAsync(image.path);
-
-        const uploadResponse = await this._imagekit.upload({
-            file,
-            fileName,
-            useUniqueFileName: this.uploadOptions.useUniqueFileName,
-            folder,
-            tags: this.uploadOptions.tags
-        })
-
-        var result = new URL(uploadResponse.url)
-        result.searchParams.append("updatedAt", new Date().getTime().toString())
-        return result.toString()
+  private urlEndpoint: string;
+  private privateKey: string;
+  private publicKey: string;
+  private enableDatedFolders: boolean;
+  private uploadOptions: ImageKitStoreConfig["uploadOptions"];
+  private _imagekit: ImageKit;
+
+  constructor(config: ImageKitStoreConfig) {
+    super();
+
+    const { auth, uploadOptions = {}, enableDatedFolders = true } = config;
+
+    const {
+      urlEndpoint = process.env.IMAGEKIT_STORE_URL_ENDPOINT,
+      privateKey = process.env.IMAGEKIT_STORE_PRIVATE_KEY,
+      publicKey = process.env.IMAGEKIT_STORE_PUBLIC_KEY,
+    } = auth || {};
+
+    const { useUniqueFileName = true, tags = [], folder = "/" } = uploadOptions;
+
+    this.enableDatedFolders =
+      enableDatedFolders.toString() === "false" || enableDatedFolders === false
+        ? false
+        : true;
+    this.urlEndpoint = urlEndpoint ?? "";
+    this.privateKey = privateKey ?? "";
+    this.publicKey = publicKey ?? "";
+    this.uploadOptions = {
+      useUniqueFileName:
+        useUniqueFileName.toString() === "false" || useUniqueFileName === false
+          ? false
+          : true,
+      tags,
+      folder: folder,
+    };
+
+    this._imagekit = new ImageKit({
+      urlEndpoint: this.urlEndpoint,
+      publicKey: this.publicKey,
+      privateKey: this.privateKey,
+    });
+  }
+
+  async exists(fileName: string, targetDir?: string): Promise<boolean> {
+    console.log("exists", fileName, targetDir);
+    try {
+      const filePath = stripLeadingSlash(join(targetDir || "", fileName));
+      const response = await got(new URL(filePath, this.urlEndpoint), {
+        responseType: "buffer",
+        resolveBodyOnly: true,
+      });
+      if (response) return true;
+    } catch (err) {
+      return false;
     }
-
-    serve() {
-        return (req: any, res: any, next: any) => {
-            console.log("serve", req.url)
-            fileServe(req, res, () => {
-                console.log("file not found")
-                res.status(404).end();
-            });
-        };
+    return false;
+  }
+
+  async save(image: StorageBase.Image, targetDir?: string): Promise<string> {
+    console.log("save", image.name);
+    try {
+      const folder =
```

Comment:
The old logic was
```
const folder = targetDir || this.getTargetDir(this.uploadOptions.folder)
```
In the new one, would `targetDir` not be used directly?

Reply:
In the old logic `this.getTargetDir(this.uploadOptions.folder)` was always called in normal flow. Now we have another flag to support dated folder structure for uploaded images.

If `targetDir` is passed to method then it will be set for folder path.